From 00bd03f70a6fd696a4d168b0d26da72b8d3b5baf Mon Sep 17 00:00:00 2001
From: Josue David Hernandez <josue.d.hernandez.gutierrez@intel.com>
Date: Wed, 2 May 2018 15:41:26 -0500
Subject: [PATCH] clr-boot-manager: add rd.luks.options=timeout=90

our systemd versions don't take rd.luks.options=timeout=0 (default)
as unlimited, it means there is not time to provide the password for an
encrypted device

Signed-off-by: Josue David Hernandez <josue.d.hernandez.gutierrez@intel.com>
---
 src/bootloaders/systemd-class.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bootloaders/systemd-class.c b/src/bootloaders/systemd-class.c
index b09497f..432273f 100644
--- a/src/bootloaders/systemd-class.c
+++ b/src/bootloaders/systemd-class.c
@@ -268,6 +268,7 @@ bool sd_class_install_kernel(const BootManager *manager, const Kernel *kernel)
         /* Add LUKS information if relevant */
         if (root_dev->luks_uuid) {
                 cbm_writer_append_printf(writer, "rd.luks.uuid=%s ", root_dev->luks_uuid);
+                cbm_writer_append_printf(writer, "rd.luks.options=timeout=90 ");
         }
 
         /* Finish it off with the command line options */
-- 
2.17.0

